actions/delete_bulk_messages_MOD.js

```diff
@@ -177,61 +209,70 @@ module.exports = {
+          times++
+          if (times === 10) throw Error('Looping for 10 times. Stop searching messages.')
+          let messages
+          if (lastId && (messagesFound.length || messagesFound.size) < limit) {
+            options.before = lastId
+            messages = await source.messages.fetch(options)
+          } else {
+            messages = await source.messages.fetch(options)
+            lastId = messages.lastKey()
+          }
+          let filtered = messages
+          if (Con0) filtered = filtered.filter(element => element.author.id != Con0.replace(/\D/g, ''))
+          if (Con1) filtered = (filtered || messages).filter(element => element.author.id == Con1.replace(/\D/g, ''))
+          if (Con2 !== '0') filtered = (filtered || messages).filter(element => (Con2 === '1') ? element.embeds.length === 0 : element.embeds.length !== 0)
+          if (Con3) filtered = (filtered || messages).filter(element => element.content.includes(Con3))
+          if (Con4) {
+            filtered = (filtered || messages).filter(function (message) {
+              let result = false
+              try {
+                result = !!eval(Con4)
+              } catch (e) {}
+              return result
+            })
+          }
+          if (Con5 !== '0') filtered = (filtered || messages).filter(element => (Con5 === '1') ? element.attachments.size === 0 : element.attachments.size !== 0)
+          messagesFound = messagesFound.concat(filtered)
         }
-        if (Con1) {
-          Con1 = Con1.replace(/\D/g, '')
-          messages = messages.filter((element) => element.author.id === Con1, this)
+        if (messagesFound.array) messagesFound = messagesFound.array()
+        if (messagesFound.length > limit) {
+          messagesFound.splice(limit)
         }
-        if (Con2 !== 0) {
-          messages = messages.filter((element) => {
-            if (Con2 === 1) {
-              return element.embeds.length === 0
-            }
-            return element.embeds.length !== 0
-          }, this)
+        const deleted = await source.bulkDelete(messagesFound)
+        const storage = parseInt(data.storage)
+        if (storage !== 0 && deleted) {
+          let result
+          (deleted.array) ? result = deleted.array() : result = deleted
+          if (deleted.length === 1) result = deleted[0]
+          const varName = this.evalMessage(data.varName2, cache)
+          this.storeValue(result, storage, varName, cache)
         }
-        if (Con3) {
-          messages = messages.filter((element) => element.content.includes(Con3), this)
+        this.callNextAction(cache)
+      } catch (err) {
+        console.error(err)
+        if (['You can only bulk delete messages that are under 14 days old.', 'Looping for 10 times. Stop searching messages.'].includes(err.message)) {
+          this.executeResults(false, data, cache)
+        } else {
+          this.displayError.bind(this, data, cache)
```

Comment:
This is not triggering the function, use [`apply()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply) or [`call()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call) instead of `bind()`